diff --git a/vendor/magento/module-configurable-product-graph-ql/Model/Cart/BuyRequest/SuperAttributeDataProvider.php b/vendor/magento/module-configurable-product-graph-ql/Model/Cart/BuyRequest/SuperAttributeDataProvider.php
index 0fa4b8da5081..daf5fffed110 100644
--- a/vendor/magento/module-configurable-product-graph-ql/Model/Cart/BuyRequest/SuperAttributeDataProvider.php
+++ b/vendor/magento/module-configurable-product-graph-ql/Model/Cart/BuyRequest/SuperAttributeDataProvider.php
@@ -93,7 +93,7 @@ public function execute(array $cartItemData): array
             throw new GraphQlNoSuchEntityException(__('Could not find specified product.'));
         }

-        $this->checkProductStock($sku, (float) $qty, (int) $cart->getStoreId());
+        $this->checkProductStock($sku, (float) $qty, (int) $cart->getStore()->getWebsite()->getId());

         $configurableProductLinks = $parentProduct->getExtensionAttributes()->getConfigurableProductLinks();
         if (!in_array($product->getId(), $configurableProductLinks)) {