diff --git a/vendor/magento/framework/View/Element/UiComponent/DataProvider/Sanitizer.php b/vendor/magento/framework/View/Element/UiComponent/DataProvider/Sanitizer.php
index 27a42fb337ce8..d49e83d1a99e0 100644
--- a/vendor/magento/framework/View/Element/UiComponent/DataProvider/Sanitizer.php
+++ b/vendor/magento/framework/View/Element/UiComponent/DataProvider/Sanitizer.php
@@ -54,7 +54,7 @@ function ($datum, string $key) use (&$config, &$toProcess) : void {
                         !is_bool($config) && !array_key_exists($key, $config)
                     )
                     && (is_string($datum) || $datum instanceof Phrase)
-                    && preg_match('/\$\{.+\}/', (string)$datum)
+                    && preg_match('/\$\{.+\}/s', (string)$datum)
                 ) {
                     //Templating is not disabled for all properties or for this property specifically
                     //Property is a string that contains template syntax so we are disabling it's rendering
