diff --git a/vendor/magento/module-catalog-graph-ql/Model/Resolver/Category/Image.php b/vendor/magento/module-catalog-graph-ql/Model/Resolver/Category/Image.php
index 549b1311..f52a7eb8 100644
--- a/vendor/magento/module-catalog-graph-ql/Model/Resolver/Category/Image.php
+++ b/vendor/magento/module-catalog-graph-ql/Model/Resolver/Category/Image.php
@@ -61,21 +61,21 @@ class Image implements ResolverInterface
             return null;
         }
         /** @var StoreInterface $store */
         $store = $context->getExtensionAttributes()->getStore();
         $baseUrl = $store->getBaseUrl(UrlInterface::URL_TYPE_WEB);
 
         $filenameWithMedia =  $this->fileInfo->isBeginsWithMediaDirectoryPath($imagePath)
             ? $imagePath : $this->formatFileNameWithMediaCategoryFolder($imagePath);
 
         if (!$this->fileInfo->isExist($filenameWithMedia)) {
-            throw new GraphQlInputException(__('Category image not found.'));
+            return null;
         }
 
         // return full url
         return rtrim($baseUrl, '/') . $filenameWithMedia;
     }
 
     /**
      * Format category media folder to filename
      *
      * @param string $fileName
