diff --git a/vendor/magento/module-sitemap/Model/Sitemap.php b/vendor/magento/module-sitemap/Model/Sitemap.php
--- a/vendor/magento/module-sitemap/Model/Sitemap.php
+++ b/vendor/magento/module-sitemap/Model/Sitemap.php
@@ -576,7 +576,9 @@
             foreach ($images->getCollection() as $image) {
                 $row .= '<image:image>';
                 $row .= '<image:loc>' . $this->_escaper->escapeUrl($image->getUrl()) . '</image:loc>';
-                $row .= '<image:title>' . $this->escapeXmlText($images->getTitle()) . '</image:title>';
+                if ($image->getTitle()) {
+                    $row .= '<image:title>' . $this->escapeXmlText($image->getTitle()) . '</image:title>';
+                }
                 if ($image->getCaption()) {
                     $row .= '<image:caption>' . $this->escapeXmlText($image->getCaption()) . '</image:caption>';
                 }
 