diff --git a/vendor/magento/module-eav/Model/Entity/AbstractEntity.php b/vendor/magento/module-eav/Model/Entity/AbstractEntity.php
index 08823bacc..6c0be486e 100644
--- a/vendor/magento/module-eav/Model/Entity/AbstractEntity.php
+++ b/vendor/magento/module-eav/Model/Entity/AbstractEntity.php
@@ -5,20 +5,21 @@
  */

 namespace Magento\Eav\Model\Entity;

 use Magento\Eav\Model\Entity\Attribute\AbstractAttribute;
 use Magento\Eav\Model\Entity\Attribute\Backend\AbstractBackend;
 use Magento\Eav\Model\Entity\Attribute\Frontend\AbstractFrontend;
 use Magento\Eav\Model\Entity\Attribute\Source\AbstractSource;
 use Magento\Eav\Model\Entity\Attribute\UniqueValidationInterface;
 use Magento\Eav\Model\ResourceModel\Attribute\DefaultEntityAttributes\ProviderInterface as DefaultAttributesProvider;
+use Magento\Framework\Api\CustomAttributesDataInterface;
 use Magento\Framework\App\Config\Element;
 use Magento\Framework\App\ObjectManager;
 use Magento\Framework\DataObject;
 use Magento\Framework\DB\Adapter\DuplicateException;
 use Magento\Framework\Exception\AlreadyExistsException;
 use Magento\Framework\Exception\LocalizedException;
 use Magento\Framework\Model\AbstractModel;
 use Magento\Framework\Model\ResourceModel\AbstractResource;
 use Magento\Framework\Model\ResourceModel\Db\ObjectRelationProcessor;
 use Magento\Framework\Model\ResourceModel\Db\TransactionManagerInterface;
@@ -1282,20 +1283,25 @@ abstract class AbstractEntity extends AbstractResource implements EntityInterfac

             if ($origData === null) {
                 $origData = [];
             }

             /**
              * drop attributes that are unknown in new data
              * not needed after introduction of partial entity loading
              */
             foreach ($origData as $k => $v) {
+                if ($k === CustomAttributesDataInterface::CUSTOM_ATTRIBUTES) {
+                    foreach ($v as $custom_attribute => $data) {
+                        $origData[$custom_attribute] = $data['value'] ?: '';
+                    }
+                }
                 if (!array_key_exists($k, $newData)) {
                     unset($origData[$k]);
                 }
             }
         } else {
             $origData = [];
         }

         $staticFields = $this->getConnection()->describeTable($this->getEntityTable());
         $staticFields = array_keys($staticFields);
