diff --git a/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Address/Attribute/Edit/Tab/General.php b/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Address/Attribute/Edit/Tab/General.php
index 3b8f7a4e8bbc..923caed8fd8a 100644
--- a/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Address/Attribute/Edit/Tab/General.php
+++ b/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Address/Attribute/Edit/Tab/General.php
@@ -187,6 +198,7 @@ protected function _prepareForm()
         );
 
         $forbiddenFileExtensions = implode(',', $this->extensionValidator->getProtectedFileExtensions());
+        $forbiddenFileExtensionsDisplay = implode(', ', $this->extensionValidator->getProtectedFileExtensions());
         $noteMessage ='Comma separated, alphanumeric characters. Forbidden file extensions: %1.';
         $fieldset->addField(
             'file_extensions',
@@ -195,7 +207,7 @@ protected function _prepareForm()
                 'name' => 'file_extensions',
                 'label' => __('File Extensions'),
                 'title' => __('File Extensions'),
-                'note' => __($noteMessage, $forbiddenFileExtensions),
+                'note' => __($noteMessage, $forbiddenFileExtensionsDisplay),
                 'class' => 'validate-forbidden-extensions',
                 'data-validation-params' => $forbiddenFileExtensions,
             ],
diff --git a/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Attribute/Edit/Tab/Main.php b/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Attribute/Edit/Tab/Main.php
index 401a9e0350c0..f04064bd4e40 100644
--- a/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Attribute/Edit/Tab/Main.php
+++ b/vendor/magento/module-customer-custom-attributes/Block/Adminhtml/Customer/Attribute/Edit/Tab/Main.php
@@ -207,6 +218,7 @@ protected function _prepareForm()
         );
 
         $forbiddenFileExtensions = implode(',', $this->extensionValidator->getProtectedFileExtensions());
+        $forbiddenFileExtensionsDisplay = implode(', ', $this->extensionValidator->getProtectedFileExtensions());
         $fieldset->addField(
             'file_extensions',
             'text',
@@ -214,7 +226,7 @@ protected function _prepareForm()
                 'name' => 'file_extensions',
                 'label' => __('File Extensions'),
                 'title' => __('File Extensions'),
-                'note' => __('Comma separated. Forbidden file extensions: %1.', $forbiddenFileExtensions),
+                'note' => __('Comma separated. Forbidden file extensions: %1.', $forbiddenFileExtensionsDisplay),
                 'class' => 'validate-forbidden-extensions',
                 'data-validation-params' => $forbiddenFileExtensions,
             ],
