diff --git a/vendor/magento/module-cms-url-rewrite-graph-ql/Model/Resolver/UrlRewrite/CmsUrlResolverIdentity.php b/vendor/magento/module-cms-url-rewrite-graph-ql/Model/Resolver/UrlRewrite/CmsUrlResolverIdentity.php
--- a/vendor/magento/module-cms-url-rewrite-graph-ql/Model/Resolver/UrlRewrite/CmsUrlResolverIdentity.php	
+++ b/vendor/magento/module-cms-url-rewrite-graph-ql/Model/Resolver/UrlRewrite/CmsUrlResolverIdentity.php	(date 1658385908000)
@@ -26,9 +26,10 @@
     public function getIdentities(array $resolvedData): array
     {
         $ids = [];
-        if (isset($resolvedData['id'])) {
+        $id = $resolvedData['id'] ?? $resolvedData['page_id'] ?? $resolvedData['identifier'] ?? null;
+        if (isset($id)) {
             $selectedCacheTag = $this->cacheTag;
-            $ids =  [$selectedCacheTag, sprintf('%s_%s', $selectedCacheTag, $resolvedData['id'])];
+            $ids =  [$selectedCacheTag, sprintf('%s_%s', $selectedCacheTag, $id)];
         }
         return $ids;
     }
