diff --git a/vendor/magento/module-sales-graph-ql/Model/Formatter/Order.php b/vendor/magento/module-sales-graph-ql/Model/Formatter/Order.php
index 06afec2a..409083b6 100644
--- a/vendor/magento/module-sales-graph-ql/Model/Formatter/Order.php
+++ b/vendor/magento/module-sales-graph-ql/Model/Formatter/Order.php
@@ -41,22 +41,21 @@ class Order
      * @throws LocalizedException
      */
     public function format(OrderInterface $orderModel): array
     {
         return [
             'created_at' => $orderModel->getCreatedAt(),
             'grand_total' => $orderModel->getGrandTotal(),
             'id' => base64_encode((string)$orderModel->getEntityId()),
             'increment_id' => $orderModel->getIncrementId(),
             'number' => $orderModel->getIncrementId(),
-            'order_date' => $this->timezone->date($orderModel->getCreatedAt())
-                ->format(DateTime::DATETIME_PHP_FORMAT),
+            'order_date' => $orderModel->getCreatedAt(),
             'order_number' => $orderModel->getIncrementId(),
             'status' => $orderModel->getStatusLabel(),
             'email' => $orderModel->getCustomerEmail(),
             'shipping_method' => $orderModel->getShippingDescription(),
             'shipping_address' => $this->orderAddress->getOrderShippingAddress($orderModel),
             'billing_address' => $this->orderAddress->getOrderBillingAddress($orderModel),
             'payment_methods' => $this->orderPayments->getOrderPaymentMethod($orderModel),
             'applied_coupons' => $orderModel->getCouponCode() ? ['code' => $orderModel->getCouponCode()] : [],
             'model' => $orderModel,
             'comments' => $this->getOrderComments($orderModel)
